spec.bs

```diff
@@ -649,19 +649,24 @@ The <dfn for=Navigator method>runAdAuction(|config|)</dfn> method steps are:
     1. Run [=update bid counts=] with |bidIgs|.
     1. Run [=interest group update=] with |auctionConfig|'s
       [=auction config/interest group buyers=].
+1. Let |generateBidResults| be a new [=list=] of [=generate bid results=].
+1. Let |auctionReporter| be a new [=auction reporter=].
 1. Let |queue| be the result of [=starting a new parallel queue=].
 1. [=parallel queue/enqueue steps|Enqueue the following steps=] to |queue|:
   1. Let |winnerInfo| be the result of running [=generate and score bids=] with |auctionConfig|,
-     null, |global|, |settings|'s [=environment/top-level origin=], and |bidIgs|.
+    null, |global|, |settings|'s [=environment/top-level origin=], |bidIgs|, |generateBidResults|,

```

Comment:
Relatedly, this method doesn't even use generateBidResults.  I suggest either:

1)  Get rid of generateBidResults here, and just use auctionReporter. Let generateBidResults be an internal detail of generate and score bids.

Or

2)  Return generateBidResults, and then use it and winnerInfo to generate auctionReporter and send reports (Maybe in another subroutine).  Could even remove the auctionReporter, and have a reporting helper method both assemble the auctionReporter and send the reports as well (Could even assemble lists on the stack, without a defined auctionReporter object).  The same helper could be used whether or not there's a winner.

Reply:
Can this be resolved?

Reply:
yes. I'll skip the "return generateBidResults instead of passing a mutable parameter" part of this comment, since it needs to be reworked in a different way as the other Issue suggests.